_core/regist/inc/sanitize.php

```diff
@@ -1,76 +1,79 @@
-        if (strlen($email) > 100)
-            return error(S_TOOLONG, $dest);
-        if (strlen($sub) > 100)
-            return error(S_TOOLONG, $dest);
-        if (strlen($resto) > 10)
-            return error(S_UNUSUAL, $dest);
-        if (strlen($url) > 10)
-            return error(S_UNUSUAL, $dest);
+    function process($post, $moderator = 0) {
+        $post['email'] = $this->CleanStr($post['email'], 0);
+        $post['email'] = preg_replace("[\r\n]", "", $post['email']);
+        $post['subject']   = $this->CleanStr($post['subject'], 0);
+        $post['subject']   = preg_replace("[\r\n]", "", $post['subject']);
+        /*$url   = $this->CleanStr($url, 0); //Or this
+        $url   = preg_replace("[\r\n]", "", $url);*/
+        $post['parent'] = $this->CleanStr($post['parent'], 0);
+        $post['parent'] = preg_replace("[\r\n]", "", $post['parent']);
+        $post['comment']   = $this->CleanStr($post['comment'], 1);
+
+        if (!$post['name'] || preg_match("/^[ |&#12288;|]*$/", $post['name']))
+            $post['name'] = "";
+        if (!$post['comment'] || preg_match("/^[ |&#12288;|\t]*$/", $post['comment']))
+            $post['comment'] = "";
+        if (!$post['subject'] || preg_match("/^[ |&#12288;|]*$/", $post['subject']))
+            $post['subject'] = "";
+
+        $post['name'] = str_replace(S_MANAGEMENT, '"' . S_MANAGEMENT . '"', $post['name']);
+        $post['name'] = str_replace(S_DELETION, '"' . S_DELETION . '"', $post['name']);
+
+        if (strlen($post['comment']) > S_POSTLENGTH)    return error(S_TOOLONG);
+        if (strlen($post['name']) > 100)                return error(S_TOOLONG);
+        if (strlen($post['email']) > 100)               return error(S_TOOLONG);
+        if (strlen($post['subject']) > 100)             return error(S_TOOLONG);
+        if (strlen($post['parent']) > 10)               return error(S_UNUSUAL);
+        //if (strlen($url) > 10)                        return error(S_UNUSUAL);
 
         // Standardize new character lines
-        $com = str_replace("\r\n", "\n", $com);
-        $com = str_replace("\r", "\n", $com);
+        $post['comment'] = str_replace("\r\n", "\n", $post['comment']);
+        $post['comment'] = str_replace("\r", "\n", $post['comment']);
 
         // Continuous lines
-        $com = preg_replace("/\n((&#12288;|)*\n){3,}/", "\n", $com);
+        $post['comment'] = preg_replace("/\n((&#12288;|)*\n){3,}/", "\n", $post['comment']);
 
-        if (!$moderator && substr_count($com, "\n") > MAX_LINES)
+        if (!$moderator && substr_count($post['comment'], "\n") > MAX_LINES)
             return error("Error: Too many lines.", $dest);
 
-        $com = nl2br($com); //br is substituted before newline char
-        $com = str_replace("\n", "", $com); //\n is erased
+        $post['comment'] = nl2br($post['comment']); //br is substituted before newline char
+        $post['comment'] = str_replace("\n", "", $post['comment']); //\n is erased
```

Comment:
42-43 can be combined into one line, `str_replace("\n", "", nl2br($post['comment']));`

Alternatively we can roll our own regsub to do `nl2br` *and* `str_replace` at once. But that's probably a bad idea.